### DISCUSSION TITLE
Lunr search results missing content

### DISCUSSION THREAD

**Discussion by adamdmharvey** [mmistakes/minimal-mistakes#1663] [Q&A] [answered]
## Environment

- Minimal Mistakes version: 4.11.2 (pinned)
- Ruby gem or remote theme version: mmistakes/minimal-mistakes@4.11.2
- Jekyll version: 3.6.2
- Git repository URL: Company Internal GitHub Enterprise
- Operating system: WIndows Ubuntu & Debian
- GitHub Pages hosted (if yes provide URL to site): Company Internal GitHub Enterprise
- Search Depth: originally default, though tested with search_full_content: true and no difference (though we confirmed that works by searching for words existing beyond line 50 in certain files)

---

## Expected behavior

Searching for a certain key word returns results in a specific file.

## Steps to reproduce the behavior

* Searching for a specific key word returns 1 result, but not for a file that contains the word in the title, excerpt, and dozens of references in the top of the content
* What we've discovered is our entire `_pages` folder, or all files at the root of the site, containing roughly a dozen or so files, do not exist in the generated lunr content store under `_site/assets/js/lunr/lunr-store.js`. Only our sub-folder collections.
* The `minimal-mistakes` docs don't describe if there is a way to include/exclude content, such as referred to by the `lunr` doc examples, so we're wondering how does mmistakes identify which content to scan? Or where to look further to troubleshoot?

**Comment by mmistakes** [comment, marked as answer]
Lunr search only i[ndexes collection documents](https://github.com/mmistakes/minimal-mistakes/blob/master/assets/js/lunr/lunr-store.js) e.g. posts or any custom collections you have. It does not currently index `site.pages` which is why you're not seeing those in search results.

I'd suggest using Jekyll Algolia instead since it's indexing is way more powerful and performant than what the theme has with Lunr.

**Comment by rdmueller** [comment]
Is it possible to add `site.pages` to the lunr index?

**Comment by rdmueller** [comment]
thanx! I've found the `assets/js/lunr/lunr-store.js` and managed to add the other pages...

**Reply by chrisjameslennon** [reply]
@rdmueller or @mmistakes  any change you could describe the tweak that was made to assets/js/lunr/lunr-store.j to enable _pages searching?  I am using github pages and don't want the extra overhead of algolia thanks!

**Comment by mrmanc** [comment]
This caught me out a bit yesterday. Just wondering whether it would be best to submit a PR to update the docs with this information, or whether a PR to include `site.pages` would be preferred? The docs currently state:

> The default search uses Lunr to build a search index of all your documents.

Which is a little misleading as it’s only posts and documents in collections.

Edit: I hadn’t spotted the incoming docs PR when I wrote this, sorry! :)

**Reply by rdmueller** [reply]
@chrisjameslennon I guess I was refering to this file https://github.com/docToolchain/docs-as-co.de/tree/master/docs/assets/js/lunr which is the source to this website: https://docs-as-co.de/

But I have to admit that I can't remember exactly what I changed.

**Reply by chrisjameslennon** [reply]
thank you @rdmueller  that helped.

For the record / anyone else, the code I ended up adding to  the mmistakes `/assets/js/lunr/lunr-store.js` file is below.  Basically just a copy and paste of what was already there, just looping through pages and adding those with a title.

```

{%- assign docs = site.pages  -%}
    
var store = [
    {%- for doc in docs -%}
      {%- if doc.header.teaser -%}
        {%- capture teaser -%}{{ doc.header.teaser }}{%- endcapture -%}
      {%- else -%}
        {%- assign teaser = site.teaser -%}
      {%- endif -%}
      {%- if doc.title -%}
      {
        "title": {{ doc.title | jsonify }},
        "excerpt":
          {%- if site.search_full_content == true -%}
            {{ doc.content | newline_to_br |
              replace:"<br />", " " |
              replace:"</p>", " " |
              replace:"</h1>", " " |
              replace:"</h2>", " " |
              replace:"</h3>", " " |
              replace:"</h4>", " " |
              replace:"</h5>", " " |
              replace:"</h6>", " "|
            strip_html | strip_newlines | jsonify }},
          {%- else -%}
            {{ doc.content | newline_to_br |
              replace:"<br />", " " |
              replace:"</p>", " " |
              replace:"</h1>", " " |
              replace:"</h2>", " " |
              replace:"</h3>", " " |
              replace:"</h4>", " " |
              replace:"</h5>", " " |
              replace:"</h6>", " "|
            strip_html | strip_newlines | truncatewords: 50 | jsonify }},
          {%- endif -%}
        "categories": {{ doc.categories | jsonify }},
        "tags": {{ doc.tags | jsonify }},
        "url": {{ doc.url | relative_url | jsonify }},
        "teaser": {{ teaser | relative_url | jsonify }}
      },
    {%- endif -%}
    {%- endfor -%}
    
  {%- for c in site.collections -%}

... etc ...
```
